### PR TITLE
Field intensity range instead of integer

### PR DESCRIPTION
**Pull request by Maleclypse**
Co-Authored-By: Zhilkin Serg <ZhilkinSerg@users.noreply.github.com>

#### Summary
None 

#### Purpose of change
Allows contributors to put in randomized fields where appropriate.  Should give more variation to chunks and map extras.

#### Describe the solution
Made the changes ZS recommended in #61702.   Then applies that solution in a couple of locations.

#### Describe alternatives you've considered
None

#### Testing
TBD

### CODE REVIEW

**Comment by NetSysFire** [comment]
Cool! Could you please also update the documentation when you have time?

**Comment by Maleclypse** [comment]
> Cool! Could you please also update the documentation when you have time?

Sure thing, when I looked last night I didn't see any documentation of intensity in mapgen so I left that part for later.

**Comment by Maleclypse** [comment]
@mqrause I think I've reverted it back to the original option ZS made.  If you have a moment could you show me the edit needed to make it accept both arrays and single integers?

File: src/mapgen.cpp [hunk 1]
```diff
@@ -1707,14 +1707,19 @@ class jmapgen_field : public jmapgen_piece
...
         time_duration age;
         bool remove;
         jmapgen_field( const JsonObject &jsi, const std::string &/*context*/ ) :
             ftype( jsi.get_member( "field" ) )
-            , intensity( jsi.get_int( "intensity", 1 ) )
             , age( time_duration::from_turns( jsi.get_int( "age", 0 ) ) )
             , remove( jsi.get_bool( "remove", false ) ) {
+            for( JsonValue jv : jsi.get_array( "intensity" ) ) {
+                intensities.push_back( jv.get_int() );
+            }
```
Comment (mqrause) [inline src/mapgen.cpp:1719]:
```suggestion
            if ( jsi.has_array( "intensity" ) ) {
                for( JsonValue jv : jsi.get_array( "intensity" ) ) {
                    intensities.push_back( jv.get_int() );
                }
            } 
```

I think doing this should be enough. This checks if the given value is an array before trying to read it. And if it's not the fallback below will be triggered to read an int.

**Pull request merged**